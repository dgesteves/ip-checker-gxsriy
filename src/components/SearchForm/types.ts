export default interface ISearchInputProps {
  onSearch: (IP: string) => void;
  searchError: string | undefined;
}
