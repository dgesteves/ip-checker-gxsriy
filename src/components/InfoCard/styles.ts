import styled from "styled-components";

export const StyledInfoCard = styled.section`
  position: absolute;
  bottom: -15%;
  z-index: 1000;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: ${({ theme }) => theme.colors.white};
  width: ${({ theme }) => `calc(${theme.sizes.s100}*10)`};
  height: ${({ theme }) => theme.sizes.s100};
  border-radius: ${({ theme }) => theme.sizes.s8};
  padding: ${({ theme }) => theme.sizes.s24};
  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    flex-direction: column;
    width: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
    height: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
    bottom: -55%;
  }
`;

export const StyledInfoCardElement = styled.div`
  display: flex;
  flex-direction: column;
  gap: ${({ theme }) => theme.sizes.s8};

  :not(:first-child) {
    border-left: 1px solid ${({ theme }) => theme.colors.gray};
    padding-left: ${({ theme }) => theme.sizes.s18};
  }

  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    align-items: center;
    :not(:first-child) {
      border-left: none;
      padding-left: 0;
    }
  }
`;

export const StyledInfoCardElementTitle = styled.h4`
  font-size: ${({ theme }) => theme.sizes.s12};
  font-weight: 500;
  text-transform: uppercase;
`;

export const StyledInfoCardElementValue = styled.h3`
  color: ${({ theme }) => theme.colors.very_dark_gray};
  font-size: ${({ theme }) => theme.sizes.s18};
  text-align: center;
`;
