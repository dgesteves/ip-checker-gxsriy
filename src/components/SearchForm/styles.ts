import styled from "styled-components";

export const StyledForm = styled.form`
  display: flex;
  justify-content: center;
  align-items: center;
  width: ${({ theme }) => `calc(${theme.sizes.s100}*5)`};
  height: ${({ theme }) => theme.sizes.s44};
  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    width: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
    touch-action: manipulation;
  }
`;

export const StyledInput = styled.input`
  width: 100%;
  height: 100%;
  border: none;
  border-top-left-radius: ${({ theme }) => theme.sizes.s8};
  border-bottom-left-radius: ${({ theme }) => theme.sizes.s8};
  padding: ${({ theme }) => theme.sizes.s16};
  font-size: ${({ theme }) => theme.sizes.s18};
  outline: none;
`;

export const StyledButton = styled.button`
  height: 100%;
  width: ${({ theme }) => theme.sizes.s44};
  border: none;
  outline: none;
  border-top-right-radius: ${({ theme }) => theme.sizes.s8};
  border-bottom-right-radius: ${({ theme }) => theme.sizes.s8};
  background-color: ${({ theme }) => theme.colors.very_dark_gray};
  cursor: pointer;
  transition: all 0.4s ease-in-out;

  :hover {
    width: ${({ theme }) => theme.sizes.s64};
  }
`;

export const StyledErrorMessage = styled.p`
  color: ${({ theme }) => theme.colors.error};
  font-size: ${({ theme }) => theme.sizes.s18};
  font-weight: 500;
  text-shadow: 0 0 80px rgb(192 219 255 / 35%), 0 0 32px rgb(65 120 255 / 24%);
`;
