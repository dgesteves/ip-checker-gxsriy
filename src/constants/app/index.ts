export const HEADER_TITLE = "IP Address Tracker";
export const SEARCH_PLACEHOLDER = "Search for any IP address or domain";
export const TIME_STANDARD = "UTC";
export const IPIFY_API_URL = "https://api.ipify.org";
export const WOOKIE_BASE_API_URL = "https://wookie.codesubmit.io/ipcheck";
export const LOCALHOST_IP = "127.0.0.1";
export const MAP_FALLBACK_TEXT = "No location found for this IP address";
export const MAP_FALLBACK_LAT = 0;
export const MAP_FALLBACK_LNG = 0;
export const MAP_ZOOM = 17;
export const VALID_IP_ADDRESS =
  /^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/;

export enum INFO_CARD_ELEMENTS {
  IP = "ip address",
  LOCATION = "location",
  TIMEZONE = "timezone",
  ISP = "isp",
}

export enum MAP_TILE_LAYER {
  ATTRIBUTION = '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors',
  URL = "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
}
