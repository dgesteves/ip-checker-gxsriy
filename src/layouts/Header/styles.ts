import styled from "styled-components";

export const StyledHeader = styled.header<{ img: string }>`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: ${({ theme }) => theme.sizes.s24};
  height: 30vh;
  padding: ${({ theme }) => theme.sizes.s24};
  background-image: url(${({ img }) => img});
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
`;

export const StyledHeaderTitle = styled.h1`
  color: ${({ theme }) => theme.colors.white};
  font-size: ${({ theme }) => theme.sizes.s28};
  font-weight: 500;
}
  `;
