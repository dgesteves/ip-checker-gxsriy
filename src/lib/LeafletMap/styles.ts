import styled from "styled-components";
import { MapContainer } from "react-leaflet";

export const StyledMapContainer = styled(MapContainer)`
  height: 70vh;
`;
