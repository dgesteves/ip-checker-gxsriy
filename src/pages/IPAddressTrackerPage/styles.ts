import styled from "styled-components";

export const StyledMapFallback = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 70vh;
`;

export const StyledMapFallbackText = styled.p`
  background-color: ${({ theme }) => theme.colors.error};
  color: ${({ theme }) => theme.colors.white};
  padding: ${({ theme }) => theme.sizes.s8} ${({ theme }) => theme.sizes.s16};
  border-radius: ${({ theme }) => theme.sizes.s8};
  font-size: ${({ theme }) => theme.sizes.s24};
  @media (max-width: ${({ theme }) => theme.breakpoints.mobile}) {
    width: ${({ theme }) => `calc(${theme.sizes.s100}*3)`};
  }
`;
