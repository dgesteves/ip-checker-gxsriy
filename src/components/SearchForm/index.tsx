import { FormEvent, useState } from "react";
import { SEARCH_PLACEHOLDER } from "../../constants/app";
import {
  StyledButton,
  StyledErrorMessage,
  StyledForm,
  StyledInput,
} from "./styles";
import ArrowIcon from "../../assets/icons/ArrowIcon";
import ISearchInputProps from "./types";

export default function SearchForm({
  onSearch,
  searchError,
}: ISearchInputProps) {
  const [searchTerm, setSearchTerm] = useState<string>("");

  const onSubmit = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    onSearch(searchTerm);
    setSearchTerm("");
  };

  return (
    <>
      <StyledForm onSubmit={onSubmit} data-testid="search-form">
        <StyledInput
          type="text"
          placeholder={SEARCH_PLACEHOLDER}
          value={searchTerm}
          onChange={(e) => setSearchTerm(e.target.value)}
        />
        <StyledButton type="submit">
          <ArrowIcon />
        </StyledButton>
      </StyledForm>
      {searchError && <StyledErrorMessage>{searchError}</StyledErrorMessage>}
    </>
  );
}
