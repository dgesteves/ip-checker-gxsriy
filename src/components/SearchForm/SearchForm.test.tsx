import { render, screen, fireEvent } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../assets/styles/theme";
import GlobalStyles from "../../assets/styles/globalStyles";
import SearchForm from "./index";
import { MAP_FALLBACK_TEXT, SEARCH_PLACEHOLDER } from "../../constants/app";

describe("SearchForm component", () => {
  const onSearch = jest.fn();
  const searchError = "";

  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <SearchForm onSearch={onSearch} searchError={searchError} />
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <SearchForm onSearch={onSearch} searchError={searchError} />
      </ThemeProvider>
    );

    expect(container).toBeInTheDocument();
  });

  it("should render the search input", () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <SearchForm onSearch={onSearch} searchError={searchError} />
      </ThemeProvider>
    );

    const inputElement = screen.getByPlaceholderText(SEARCH_PLACEHOLDER);

    expect(inputElement).toBeInTheDocument();
    expect(inputElement).toHaveAttribute("type", "text");
  });

  it("should have correct value on search input change", async () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <SearchForm onSearch={onSearch} searchError={searchError} />
      </ThemeProvider>
    );

    const inputElement = screen.getByPlaceholderText(SEARCH_PLACEHOLDER);

    expect(inputElement).toHaveValue("");

    fireEvent.change(inputElement, { target: { value: "test" } });

    expect(inputElement).toHaveValue("test");
  });

  it("should render the search button", () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <SearchForm onSearch={onSearch} searchError={searchError} />
      </ThemeProvider>
    );

    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  it("should call onSearch function when search button is clicked", async () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <SearchForm onSearch={onSearch} searchError={searchError} />
      </ThemeProvider>
    );

    expect(onSearch).not.toHaveBeenCalled();

    const searchButton = screen.getByRole("button");

    fireEvent.click(searchButton);

    expect(onSearch).toHaveBeenCalled();
  });

  it("should render the search error", () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <SearchForm onSearch={onSearch} searchError={MAP_FALLBACK_TEXT} />
      </ThemeProvider>
    );

    expect(screen.getByText(MAP_FALLBACK_TEXT)).toBeInTheDocument();
  });

  it("should render the search error with correct error message", () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <SearchForm onSearch={onSearch} searchError={MAP_FALLBACK_TEXT} />
      </ThemeProvider>
    );

    const pElement = screen.getByText(MAP_FALLBACK_TEXT);

    expect(pElement.textContent).toBe(MAP_FALLBACK_TEXT);
  });
});
