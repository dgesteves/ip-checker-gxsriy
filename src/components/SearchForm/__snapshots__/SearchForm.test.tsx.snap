// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component renders correctly 1`] = `
<div>
  <form
    class="sc-bczRLJ fnJLah"
    data-testid="search-form"
  >
    <input
      class="sc-gsnTZi kUkneN"
      placeholder="Search for any IP address or domain"
      type="text"
      value=""
    />
    <button
      class="sc-dkzDqf fsoqdr"
      type="submit"
    >
      <svg
        height="14"
        width="11"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m2 1 6 6-6 6"
          fill="none"
          stroke="#fff"
          stroke-width="3"
        />
      </svg>
    </button>
  </form>
</div>
`;
